tests/testthat/test-expr-expr.R

```diff
@@ -29,3 +29,2946 @@ test_that("map_batches works", {
+      n0 = pl$col("a")$pct_change(0)
+    ),
+    pl$DataFrame(
+      n1 = r_pct_chg(l$a),
+      n2 = r_pct_chg(l$a, n = 2),
+      n0 = r_pct_chg(l$a, n = 0)
+    )
+  )
+})
+
+test_that("skew", {
+  R_skewness <- function(x, bias = TRUE, na.rm = FALSE) {
+    if (na.rm) x <- x[!is.na(x)]
+    n <- length(x)
+    m2 <- sum((x - mean(x))^2) / n
+    m3 <- sum((x - mean(x))^3) / n
+    biased_skewness <- m3 / m2^(3 / 2)
+    if (bias) {
+      biased_skewness
+    } else {
+      correction <- sqrt(n * (n - 1L)) / (n - 2)
+      biased_skewness * correction
+    }
+  }
+
+  l <- list(a = c(1:3, 2:1), b = c(1:3, NA_integer_, 1L))
+  expect_equal(
+    pl$DataFrame(!!!l)$select(
+      a_skew = pl$col("a")$skew(),
+      a_skew_bias_F = pl$col("a")$skew(bias = FALSE),
+      b_skew = pl$col("b")$skew(),
+      b_skew_bias_F = pl$col("b")$skew(bias = FALSE)
+    ),
+    pl$DataFrame(
+      a_skew = R_skewness(l$a),
+      a_skew_bias_F = R_skewness(l$a, bias = FALSE),
+      b_skew = R_skewness(l$b, na.rm = TRUE),
+      b_skew_bias_F = R_skewness(l$b, bias = FALSE, na.rm = TRUE)
+    )
+  )
+})
+
+test_that("kurtosis", {
+  R_kurtosis <- function(x, fisher = TRUE, bias = TRUE, na.rm = TRUE) {
+    if (na.rm) x <- x[!is.na(x)]
+    n <- length(x)
+    m2 <- sum((x - mean(x))^2) / n
+    m4 <- sum((x - mean(x))^4) / n
+    fisher_correction <- if (fisher) 3 else 0
+    biased_kurtosis <- m4 / m2^2
+    if (bias) {
+      biased_kurtosis - fisher_correction
+    } else {
+      correction <- 1.0 / (n - 2) / (n - 3) * ((n**2 - 1.0) * m4 / m2**2.0 - 3 * (n - 1)**2.0)
+      correction + 3 - fisher_correction
+    }
+  }
+
+  l <- list(a = c(1:3, NA_integer_, 1:3))
+  l2 <- list(a = c(1:3, 1:3))
+
+  # missing values should not change outcome
+  expect_equal(
+    pl$DataFrame(!!!l)$select(
+      kurt = pl$col("a")$kurtosis(),
+      kurt_TF = pl$col("a")$kurtosis(fisher = TRUE, bias = FALSE),
+      kurt_FT = pl$col("a")$kurtosis(fisher = FALSE, bias = TRUE),
+      kurt_FF = pl$col("a")$kurtosis(fisher = FALSE, bias = FALSE)
+    ),
+    pl$DataFrame(!!!l2)$select(
+      kurt = pl$col("a")$kurtosis(),
+      kurt_TF = pl$col("a")$kurtosis(fisher = TRUE, bias = FALSE),
+      kurt_FT = pl$col("a")$kurtosis(fisher = FALSE, bias = TRUE),
+      kurt_FF = pl$col("a")$kurtosis(fisher = FALSE, bias = FALSE)
+    )
+  )
+
+  # equivalence with R
+  expect_equal(
+    pl$DataFrame(!!!l2)$select(
+      kurt_TT = pl$col("a")$kurtosis(),
+      kurt_TF = pl$col("a")$kurtosis(fisher = TRUE, bias = FALSE),
+      kurt_FT = pl$col("a")$kurtosis(fisher = FALSE, bias = TRUE),
+      kurt_FF = pl$col("a")$kurtosis(fisher = FALSE, bias = FALSE)
+    ),
+    pl$DataFrame(
+      kurt_TT = R_kurtosis(l2$a, TRUE, TRUE),
+      kurt_TF = R_kurtosis(l2$a, TRUE, FALSE),
+      kurt_FT = R_kurtosis(l2$a, FALSE, TRUE),
+      kurt_FF = R_kurtosis(l2$a, FALSE, FALSE)
+    )
+  )
+})
+
+# TODO-REWRITE: panicks when one of the bounds is NaN
```

Comment:
Don't know why this occurs, it throws a normal error in python

Reply:
Sorry, but this test is quite hard to read. I don't know what you are trying to do. Could you please rewrite it?